scripts/events_module/event_filters.py

```diff
@@ -0,0 +1,412 @@
+import re
+
+import ujson
+
+from scripts.game_structure.game_essentials import game
+from scripts.utility import (
+    get_alive_status_cats,
+    filter_relationship_type,
+)
+
+
+def event_for_location(locations: list) -> bool:
+    """
+    checks if the clan is within the given locations
+    """
+    if "any" in locations:
+        return True
+
+    for place in locations:
+        if ":" in place:
+            info = place.split(":")
+            req_biome = info[0]
+            req_camps = info[1].split("_")
+        else:
+            req_biome = place
+            req_camps = ["any"]
+
+        if req_biome == game.clan.biome.lower():
+            if "any" in req_camps or game.clan.camp_bg in req_camps:
+                return True
+            else:
+                return False
+            
+        return False
+
+
+def event_for_season(seasons: list) -> bool:
+    """
+        checks if the clan is within the given seasons
+        """
+    if "any" in seasons:
+        return True
+    elif game.clan.current_season.lower() in seasons:
```

Comment:
```suggestion
    if "any" in seasons or game.clan.current_season.lower() in seasons:
```
Python is lazy in its evaluation: if the first part of an `or` is true, it doesn't bother evaluating the second and just proceeds. So you can make this shorter again!